editor/metaboxes/iframe.js

```diff
@@ -0,0 +1,291 @@
+			/**
+			 * Sets up an event listener for resizing. The resizing occurs inside
+			 * the iframe, see gutenberg/assets/js/metabox.js
+			 */
+			window.addEventListener( 'message', this.checkMessageForResize, false );
+
+			// Initially set node to not display anything so that when it loads, we can see it.
+			this.node.style.display = 'none';
+			this.node.addEventListener( 'load', this.observeChanges );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		// Metabox updating.
+		if ( this.props.isUpdating === false && nextProps.isUpdating === true ) {
+			const iframe = findDOMNode( this.node );
+
+			//iframe.addEventListener( 'load', this.handleMetaboxReload );
+
+			this.clonedNode = iframe.cloneNode( true );
+			this.hideNode( this.clonedNode );
+			const parent = iframe.parentNode;
+
+			parent.appendChild( this.clonedNode );
+
+			/**
+			 * When the dom content has loaded for the cloned iframe handle the
+			 * double buffering.
+			 */
+			this.clonedNode.addEventListener( 'load', this.handleDoubleBuffering );
+		}
+	}
+
+	componentDidUpdate() {
+		if ( this.isFrameAccessible() ) {
+			/**
+			 * Sets up an event listener for resizing. The resizing occurs inside
+			 * the iframe, see gutenberg/assets/js/metabox.js
+			 */
+			window.addEventListener( 'message', this.checkMessageForResize, false );
+
+			// Initially set node to not display anything so that when it loads, we can see it.
+			//this.node.style.display = 'none';
+			this.node.addEventListener( 'load', this.observeChanges );
+		}
+	}
+
+	handleDoubleBuffering() {
+		const iframe = findDOMNode( this.node );
+		const cloneIframe = findDOMNode( this.clonedNode );
+		// The standard post.php form ID post should probably be mimicked.
+		const form = iframe.contentWindow.document.getElementById( 'post' );
+
+		form.submit();
+
+		const cloneForm = cloneIframe.contentWindow.document.getElementById( 'post' );
+
+		cloneForm.parentNode.replaceChild( form, cloneForm );
+
+		this.showNode( this.clonedNode );
+		this.hideNode( this.node );
+
+		this.node.addEventListener( 'load', this.handleMetaboxReload );
+	}
+
+	hideNode( node ) {
+		node.style.visibility = 'hidden';
+		node.style.position = 'absolute';
+		node.style.top = '0';
+		node.style.left = '0';
+	}
+
+	showNode( node ) {
+		node.style.visibility = null;
+		node.style.position = null;
+		node.style.top = null;
+		node.style.left = null;
+	}
+
+	componentWillUnmount() {
+		if ( this.isFrameAccessible() ) {
+			const iframe = findDOMNode( this.node );
+			iframe.removeEventListener( 'message', this.checkMessageForResize );
+
+			if ( this.dirtyObserver ) {
+				this.dirtyObserver.disconnect();
+			}
+
+			if ( this.form !== null ) {
+				this.form.removeEventListener( 'input', this.checkMetaboxState );
+				this.form.removeEventListener( 'change', this.checkMetaboxState );
+			}
+
+			this.node.removeEventListener( 'load', this.observeChanges );
+		}
+	}
+
+	observeChanges() {
+		const node = findDOMNode( this.node );
```

Comment:
> This question also applies to some other uses of findDOMNode( this.node ).

For whatever reason this would not work without findDOMNode, potentially when React is bypassed the ref goes stale or something. Will definitely work on cleaning this up.

Reply:
In case [this](https://facebook.github.io/react/docs/refs-and-the-dom.html) helps:

> React will call the ref callback with the DOM element when the component mounts, _and call it with null when it unmounts._

and another caveat:

> If the ref callback is defined as an inline function, it will get called twice during updates, first with null and then again with the DOM element. This is because a new instance of the function is created with each render, so React needs to clear the old ref and set up the new one. You can avoid this by defining the ref callback as a bound method on the class, but note that it shouldn't matter in most cases.